### PR TITLE
Feat: Add json support

### LINKED ISSUE #163: Support JSON inputs

### PR DESCRIPTION
**Pull request by nipierre**
Fixes #163 .

- [x] JSON support in decoding.
- [x] Bump `stainless_ffmpeg` to close memory leak when decoding.
- [x] Refactored MediaStream with simplified buffer handling.

### CODE REVIEW

File: rs_mcai_worker_sdk/src/message/media/json/decoder.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        );
+
+        self.buffer.push_back(actual_json.to_string());
+        log::trace!("Incomplete JSON content added to buffer: {}", actual_json);
+        return Ok(None);
+      };
+
+      let value: Value = result.unwrap();
+
+      if value.is_object() {
```
Comment (nipierre) [inline rs_mcai_worker_sdk/src/message/media/json/decoder.rs]: Invert condition

File: rs_mcai_worker_sdk/src/message/media/mod.rs [hunk 2]
```diff
@@ -49,6 +54,28 @@ impl StreamDescriptor {
       index,
       audio_configuration: None,
       image_configuration: Some(ImageConfiguration { filters }),
+      ebu_ttml_live_configuration: None,
+      json_configuration: None,
+    }
+  }
+
+  pub fn new_ebu_ttml_live(index: usize) -> Self {
+    StreamDescriptor {
```
Comment (nipierre) [inline rs_mcai_worker_sdk/src/message/media/mod.rs:64]: Modify stream descriptor with 2 entries :
 - index
 - configuration, which is an enum running on the different configuration possible (image, audio, EBU, JSON , Data)

File: rs_mcai_worker_sdk/src/message/media/mod.rs [hunk 3]
```diff
@@ -27,36 +28,56 @@ pub const DESTINATION_PATH_PARAMETER: &str = "destination_path";
...
 pub const STOP_INDEX_PARAMETER: &str = "sdk_stop_index";
 
+#[cfg(all(feature = "media"))]
+#[derive(Debug, PartialEq)]
+pub enum StreamConfiguration {
+  Audio(Option<AudioConfiguration>),
+  Image(Option<ImageConfiguration>),
+  EbuTtmlLive(),
+  Json(),
+  Data(),
```
Comment (valnoel) [inline rs_mcai_worker_sdk/src/message/media/mod.rs]: Are the empty brackets necessary?

File: rs_mcai_worker_sdk/src/message/media/mod.rs [hunk 4]
```diff
@@ -27,36 +28,56 @@ pub const DESTINATION_PATH_PARAMETER: &str = "destination_path";
 pub const START_INDEX_PARAMETER: &str = "sdk_start_index";
 pub const STOP_INDEX_PARAMETER: &str = "sdk_stop_index";
 
+#[cfg(all(feature = "media"))]
+#[derive(Debug, PartialEq)]
+pub enum StreamConfiguration {
+  Audio(Option<AudioConfiguration>),
+  Image(Option<ImageConfiguration>),
```
Comment (valnoel) [inline rs_mcai_worker_sdk/src/message/media/mod.rs]: Are the `Option`s really useful?

File: rs_mcai_worker_sdk/src/message/media/source.rs [hunk 5]
```diff
@@ -34,6 +31,14 @@ use std::{
   thread,
 };
 
+pub enum DecoderType {
+  Audio(Option<AudioDecoder>),
+  Video(Option<VideoDecoder>),
+  EbuTtmlLive(Option<EbuTtmlLiveDecoder>),
+  Json(Option<JsonDecoder>),
```
Comment (valnoel) [inline rs_mcai_worker_sdk/src/message/media/source.rs]: Are the `Option`s really useful?

File: rs_mcai_worker_sdk/src/message/media/source.rs [hunk 6]
```diff
@@ -34,6 +31,14 @@ use std::{
...
+  Data(),
```
Comment (valnoel) [inline rs_mcai_worker_sdk/src/message/media/source.rs]: Are the empty brackets necessary?

File: rs_mcai_worker_sdk/src/message/media/json/decoder.rs [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    log::debug!("Try decoding: {}", json_content);
+
+    let json_value = self.decode_content(&json_content)?;
+
+    forget(json_content);
+
+    Ok(json_value)
+  }
+
+  pub fn decode_content(&mut self, json_content: &str) -> Result<Option<Json>, String> {
```
Comment (valnoel) [inline rs_mcai_worker_sdk/src/message/media/json/decoder.rs:36]: Could you add unit tests? On this function at least...

**Review by valnoel — CHANGES_REQUESTED** [review]

File: rs_mcai_worker_sdk/src/message/media/json/decoder.rs (hunk 7 above)
Comment (nipierre) [reply rs_mcai_worker_sdk/src/message/media/json/decoder.rs:36]: Done for the decoding function, for the rest we probably should implement a whole Packet generator.. :/

**Review by valnoel — APPROVED** [review]

**Pull request merged**